nereid/nereid/src/nomograph/interpolators.py

```diff
@@ -0,0 +1,382 @@
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy
+import pandas
+from matplotlib.axes import Axes
+from scipy.interpolate import CloughTocher2DInterpolator as CT2DI
+
+
+def bisection_search(
```

Comment:
the kicker is that we have to transform f so that we ensure a and b have different signs before we could use `optimize.bisect`, but in this case our function is a slice of a 2dcloughtocher interpolation, and i don't know how to transform it such that the value i 'know' is =0 for the value i want back.

if this can be made to work with a scipy built in, i'd actually prefer it, but for now it converges in 4-8 iterations, so it's pretty fast for this use case.